### PR TITLE
Add bee types (drone/princess/queen) and dynamic item icons and names

### PR DESCRIPTION
**Pull request by PS-Soundwave**
Adds the last missing features from #1.

Testing should be similar to #1, with the following additional points:
- The instructions for #1 assume no mutations. You may use species A and D for this.
- 'beeMateTarget' should now only target princesses (red bees)
- 'beeMate' should now only target drones (white bees)
- A mated bee is now a queen (which should be blue)
- Bees should now show their primary species (that is, the species indicated by the first active gene, with 1 -> A, 2 -> B, etc.) on the icon, rather than offspring always being As.
- All of these should show appropriate tooltips of the form "<species> Drone/Princess/Queen"
- Count for 'beeBirth' now specifies drone offspring count. A princess should always be birthed.

### CODE REVIEW

File: src/main/java/org/terasology/projsndwv/TempBeeRegistry.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * A temporary class providing helpers intended to be provided by a future bee registry.
+ */
+public class TempBeeRegistry {
+    @In
+    public static AssetManager assetManager;
+
+    public static TextureRegionAsset<?> getTextureRegionAssetForSpeciesAndType(int species, int type) {
+        return (TextureRegionAsset<?>)CoreRegistry.get(AssetManager.class).getAsset("Apiculture:bee_" + new String[] {"a", "b", "c", "d"}[species - 1] + "_" + new String[] {"drone", "princess", "queen"}[type], TextureRegionAsset.class).get();
```
Comment (syntaxi) [inline src/main/java/org/terasology/projsndwv/TempBeeRegistry.java:32]: I get that it's a temporary function in a temporary class, but this still gives me the heebie jeebies. It's fine so long as you remove it later on.

File: src/main/java/org/terasology/projsndwv/components/BeeComponent.java [hunk 2]
```diff
@@ -17,5 +17,14 @@
...
+public final class BeeComponent implements Component {
+    // TODO: Make enum
+    /**
+     * Bee type, that is, whether a bee is a drone, princess, or queen.<br/>
+     * <br/>
+     * 0 - Drone<br/>
+     * 1 - Princess<br/>
+     * 2 - Queen
+     */
+    public int type;
```
Comment (syntaxi) [inline src/main/java/org/terasology/projsndwv/components/BeeComponent.java]: I would prefer to see this as an enum for clarity.

File: src/main/java/org/terasology/projsndwv/commands/BeeCommands.java [hunk 3]
```diff
@@ -146,17 +158,27 @@ public String birth(@Sender EntityRef client, @CommandParam(value = "count", req
         }
 
         EntityRef item = client.getComponent(ClientComponent.class).character.getComponent(CharacterHeldItemComponent.class).selectedItem;
-        if (item.getComponent(BeeComponent.class) == null || item.getComponent(MatedComponent.class) == null) {
-            return "Held item is not a mated bee.";
+        if (item.getComponent(BeeComponent.class) == null || item.getComponent(MatedComponent.class) == null || item.getComponent(BeeComponent.class).type != 2) {
```
Comment (syntaxi) [inline src/main/java/org/terasology/projsndwv/commands/BeeCommands.java]: Prefer either
```java
BeeComponent component = item.getComoponent(BeeComponent.class);
if (component == null) {
    // Stuff
}
```
Or
```java
if (!item.hasComponent(BeeComponent.class) {
  //Stuff
}
```

Obviously whichever fits your style and code best but `hasComponent` is there for a reason

File: src/main/java/org/terasology/projsndwv/commands/BeeCommands.java [hunk 4]
```diff
@@ -146,17 +158,27 @@ public String birth(@Sender EntityRef client, @CommandParam(value = "count", req
...
 
-        for (int i = 0; i < count; i++) {
-            EntityRef drop = entityManager.create("Apiculture:bee");
+        EntityRef drop = entityManager.create("Apiculture:bee_princess");
+        drop.addComponent(offspringGenetics.next());
+        ItemComponent itemComponent = drop.getComponent(ItemComponent.class);
+        itemComponent.icon = TempBeeRegistry.getTextureRegionAssetForSpeciesAndType(drop.getComponent(GeneticsComponent.class).activeGenes.get(0), 1);
+        drop.addComponent(itemComponent);
+        drop.addComponent(TempBeeRegistry.getDisplayNameComponentForSpeciesAndType(drop.getComponent(GeneticsComponent.class).activeGenes.get(0), 1));
+        drop.send(new DropItemEvent(client.getComponent(ClientComponent.class).character.getComponent(LocationComponent.class).getWorldPosition()));
```
Comment (syntaxi) [inline src/main/java/org/terasology/projsndwv/commands/BeeCommands.java:180]: Maybe see if you can place the items directly into the players inventory like the `give` command does.
Again, it's a temp command so if it's no worth it, that's fine

File: src/main/java/org/terasology/projsndwv/components/BeeComponent.java (hunk 2 above)
Comment (PS-Soundwave) [reply src/main/java/org/terasology/projsndwv/components/BeeComponent.java]: @jellysnake Mind directing me to an example of an enum being used in prefabs? I don't know how prefabs initialize enum fields, how the constructor must be formatted, etc.

File: src/main/java/org/terasology/projsndwv/commands/BeeCommands.java (hunk 4 above)
Comment (PS-Soundwave) [reply src/main/java/org/terasology/projsndwv/commands/BeeCommands.java:180]: This requires a dependency on Inventory. I believe I'll have to pull Inventory in later anyways, but that was the motivation for dropping in world as of present.

**Comment by Cervator** [comment]
Can confirm now that it worked for me without errors or weird stacktraces 👍 

Hard to glean a thorough meaning from it without being a geneticist or having more gameplay affecting elements to it, but I can see that things happen :-)

**Pull request merged**